src/DoctrineWriter.php

```diff
@@ -53,32 +59,48 @@ class DoctrineWriter implements Writer, Writer\FlushableWriter
     protected $truncate = true;
 
     /**
-     * List of fields used to lookup an entity
+     * List of fields used to lookup an object
      *
      * @var array
      */
     protected $lookupFields = array();
 
     /**
-     * @param EntityManagerInterface $entityManager
-     * @param string                 $entityName
-     * @param string|array           $index Field or fields to find current entities by
+     * Constructor
+     *
+     * @param ObjectManager $objectManager
+     * @param string        $objectName
+     * @param string|array        $index         Field or fields to find current entities by
      */
-    public function __construct(EntityManagerInterface $entityManager, $entityName, $index = null)
+    public function __construct(ObjectManager $objectManager, $objectName, $index = null)
     {
-        $this->entityManager = $entityManager;
-        $this->entityRepository = $entityManager->getRepository($entityName);
-        $this->entityMetadata = $entityManager->getClassMetadata($entityName);
-        //translate entityName in case a namespace alias is used
-        $this->entityName = $this->entityMetadata->getName();
-        if ($index) {
-            if (is_array($index)) {
+        $this->ensureSupportedObjectManager($objectManager);
+        $this->objectManager = $objectManager;
+        $this->objectRepository = $objectManager->getRepository($objectName);
+        $this->objectMetadata = $objectManager->getClassMetadata($objectName);
+        //translate objectName in case a namespace alias is used
+        $this->objectName = $this->objectMetadata->getName();
+        if($index) {
+            if(is_array($index)) {
                 $this->lookupFields = $index;
             } else {
                 $this->lookupFields = [$index];
             }
         }
     }
+    
+    protected function ensureSupportedObjectManager(ObjectManager $objectManager)
+    {
+        if (!($objectManager instanceof \Doctrine\ORM\EntityManager
+            || $objectManager instanceof \Doctrine\ODM\MongoDB\DocumentManager)
+        ) {
+            $message = sprintf(
```

Comment:
Can the construction logic be moved to the exception? Either by overriding the constructor or creating a static named constructor. But I don't think there are many use cases for this exception, so you could just stick to option 1.